src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_ARRAY_SIZE = 10;
+    private static final double RESIZE_FACTOR = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[INITIAL_ARRAY_SIZE];
+        this.size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        expandCapacityIfArrayFull();
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, true);
```

Comment:
we need specific method for check for add. using booleans as method parameters is a bad practice. never be worry about creating separate method in such case
```suggestion
        checkIndexForAdd(index);
```